### PR TITLE
hack/Makefile: detect dirty vendor

### PR DESCRIPTION
**Pull request by fuweid**
use containerd/project script to detect dirty vendor.

close: #1012

Signed-off-by: Wei Fu <fuweid89@gmail.com>

### CODE REVIEW

**Comment by fuweid** [comment]
the k8s-ci workflow fails because of linter issue. wait for #1027.

**Comment by Random-Liu** [comment]
@fuweid Can we add a `hack/verify-vendor.sh` and call it in `hack/verify.sh`.

I want this to be part of `pull-cri-containerd-verify` job, not travis job. :)

File: hack/verify-vendor.sh [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+set -o errexit
+set -o nounset
+set -o pipefail
+
+tmpdir="$(mktemp -d)"
+trap "rm -rf ${tmpdir}" EXIT
+
+git clone "https://github.com/containerd/project" "${tmpdir}"
+"${tmpdir}"/script/validate/vendor
+unset tmpdir
```
Comment (Random-Liu) [inline hack/verify-vendor.sh]: If you unset `tmpdir` here, does trap still work?

Comment (Random-Liu) [reply hack/verify-vendor.sh]: Hm, it still works, interesting.

We don't need to unset it actually, but it works anyway. :)

Comment (fuweid) [reply hack/verify-vendor.sh]: :) I use `"` instead of `'`. I can remove the `unset` for the case, since it confuses.

**Pull request merged**